doc/datasets/sample_generators.rst

```diff
@@ -29,10 +29,56 @@ per class; and linear transformations of the feature space.
 near-equal-size classes separated by concentric hyperspheres.
 :func:`make_hastie_10_2` generates a similar binary, 10-dimensional problem.
 
-.. image:: ../auto_examples/datasets/images/sphx_glr_plot_random_dataset_001.png
-   :target: ../auto_examples/datasets/plot_random_dataset.html
+.. plot::
+   :context: close-figs
    :scale: 50
    :align: center
+   :include-source: False
```

Comment:
The code snippet is a bit long if placed here, so I just hide the code. WDYT @glemaitre ?